### PR TITLE
LGPU simulator support controlled ops

### PR DESCRIPTION
**Pull request by josephleekl**
### Before submitting

Please complete the following checklist when submitting a PR:

- [X] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [X] Ensure that the test suite passes, by running `make test`.

- [X] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [X] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
This PR updates the Lightning GPU simulator for Catalyst to support controlled gates and matrices. 

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-79061]

### CODE REVIEW

File: pennylane_lightning/core/src/simulators/lightning_gpu/catalyst/tests/Test_LightningGPUSimulator.cpp [hunk 1]
```diff
@@ -678,6 +678,95 @@ TEST_CASE("LightningGPUSimulator::GateSet", "[GateSet]") {
         REQUIRE(LGPUsim->CacheManagerInfo() == expected);
     }
 
+    // ============= Controlled operations =============
```
Comment (multiphaseCFD) [inline pennylane_lightning/core/src/simulators/lightning_gpu/catalyst/tests/Test_LightningGPUSimulator.cpp:681]: Do we want to remove this line?

**Review by multiphaseCFD — APPROVED** [review]
LGTM!Thanks @josephleekl !

**Review by AmintorDusko — APPROVED** [review]
Looks good!

File: pennylane_lightning/core/src/simulators/lightning_gpu/catalyst/tests/Test_LightningGPUSimulator.cpp (hunk 1 above)
Comment (josephleekl) [reply pennylane_lightning/core/src/simulators/lightning_gpu/catalyst/tests/Test_LightningGPUSimulator.cpp:681]: This is in a big test and helpful to mark the specific areas. Also similar to [this line ](https://github.com/PennyLaneAI/pennylane-lightning/blob/422fc7966a001496c47ea8b3776ce65cc0062f0e/pennylane_lightning/core/src/simulators/lightning_gpu/catalyst/tests/Test_LightningGPUSimulator.cpp#L318) in the file

**Pull request merged**